lib/src/inherited_provider.dart

```diff
@@ -694,14 +694,15 @@ class _CreateInheritedProviderState<T>
   bool _didSucceedInit = false;
   T? _value;
   _CreateInheritedProvider<T>? _previousWidget;
+  FlutterErrorDetails? _initError;
 
   @override
   T get value {
     if (_didInitValue && !_didSucceedInit) {
       throw StateError(
```

Comment:
Now that rethrowing an error with a specific stacktrace is a thing (Error.throwWithStackTrace), maybe we should replacd the StateError in favor of throwing the original error?

Reply:
The currents tests are explicitly validating that there is a state error:

https://github.com/rrousselGit/provider/blob/0bbe2269a79e3c50e611cb9b29d38cf120dee77a/test/inherited_provider_test.dart#L2406

I don't know if it would make sense to catch the `StateError` somewhere, but if somebody does, this would be a breaking change.

----

I have looked up [`Error.throwWithStackTrace`](https://api.dart.dev/dev/2.16.0-9.0.dev/dart-core/Error/throwWithStackTrace.html) and it turns out that this method will be introduced with Dart version 2.16. The Flutter stable channel is currently using Dart version 2.15.1. The Flutter dev channel is using Dart version 2.16.0-80.1.beta.

----------

So, if you ask for my opinion, I would stick to throwing the `StateError`, at least until the Flutter stable channel uses Dart 2.16.

Even then, it would be a rather cosmetic change for a rather unusual edge case. So, even though I think that `Error.throwWithStackTrace` may be more elegant, I personally would not do this change, just because of the required effort.

----------

If I may rant a little, I am very surprise that Dart exceptions do not have a `cause` parameter for exceptions, like I am used to from Java. For me the best solution would be pass the `_initError` as `cause` parameter into the `StateError`. Then the Dart VM can figure out how to print it to the console.

Reply:
No problem, this can wait